MapboxCoreNavigation/FreeDriveLocationManager.swift

```diff
@@ -0,0 +1,134 @@
+import MapboxNavigationNative
+import CoreLocation
+import MapboxDirections
+import MapboxAccounts
+
+public protocol FreeDriveDebugInfoListener: AnyObject {
+    func didGet(location: CLLocation, with matches: [MapMatch], for rawLocation: CLLocation)
+}
+
+open class FreeDriveLocationManager: NavigationLocationManager, CLLocationManagerDelegate {
+    public weak var debugInfoListener: FreeDriveDebugInfoListener? {
+        get {
+            proxyDelegate?.debugDelegate
+        }
+        set {
+            proxyDelegate?.debugDelegate = newValue
+        }
+    }
+
+    private var proxyDelegate: ProxyDelegate?
+
+    public required init(navigator: Navigator? = nil) {
+        proxyDelegate = ProxyDelegate()
+        super.init()
+
+        if let navigator = navigator {
+            proxyDelegate?.navNative = navigator
+        } else {
+            let tilesVersion = RouteTilesVersion(with: Directions.shared.credentials)
```

Comment:
The developer needs to be able to specify a Directions object other than the shared Directions object, in case they need to specify an access token programmatically.